contrib/devtools/fix-copyright-headers.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+current_year=$(date +%Y)
+
+read -r -d '' VERIFY_SCRIPT << EOS
+for party in "The Zcash developers" "The Bitcoin Core developers" "Bitcoin Developers"; do
+  sed -i"" -e "s#Copyright (c) \([0-9]\{4\}\)\(-[0-9]\{4\}\)\? \$party#Copyright (c) \1-$current_year \$party#" COPYING
+  sed -i"" -e "s#\(.*\)\([0-9]\{4\}\)\(-[0-9]\{4\}\)\, \$party#\1\2-$current_year, \$party#" contrib/debian/copyright
+done
+
+sed -i"" -e "s/define(_COPYRIGHT_YEAR, [0-9]\{4\})/define(_COPYRIGHT_YEAR, $current_year)/" configure.ac
+sed -i"" -e "s/#define COPYRIGHT_YEAR [0-9]\{4\}/#define COPYRIGHT_YEAR $current_year/" src/clientversion.h
+
+git grep "^// Copyright (c) .* The Zcash developers" \\
```

Comment:
Nit: this is updating _every_ occurrence of the header, even on files that have not been otherwise altered. I thought that the intention was to only bump the year for files that were modified in 2023, but looking at #5950 again I see that what I was thinking of was that _missing_ Zcash copyright headers were added to files from upstream that had been altered by us, and we just set all of the end dates to 2022. I guess it doesn't matter as long as we are internally consistent for the Zcash copyright headers (and we can always switch later).